src/Components/Toast.jsx

```diff
@@ -26,51 +27,71 @@ function Toast({ message, handleMessageRemove }) {
       clearTimeout(timeoutIdForRemoval);
       clearTimeout(timeoutIdForAnimation);
     };
-  }, []);
+  }, [handleMessageRemove, message.id]);
 
   const styles = {};
 
+  const handleCloseBtn = useCallback(() => {
+    setSlideOut(true);
+    setTimeout(() => {
+      handleMessageRemove(message.id);
+    }, 450);
+  }, [message.id, handleMessageRemove]);
+
   const toastIcon = () => {
-    if (message.toastType === "notice") {
-      styles.backgroundColor = "#D69AFF";
-      return <FiAlertOctagon className="toast-icon" />;
-    } else if (message.toastType === "warning") {
-      styles.backgroundColor = "#CECE4A";
-      return <FiTool className="toast-icon" />;
-    } else if (message.toastType === "success") {
-      styles.backgroundColor = "#5FD65F";
-      return <FiCheckCircle className="toast-icon" />;
-    } else if (message.toastType === "error") {
-      styles.backgroundColor = "#BB4D4D";
-      return <FiAlertTriangle className="toast-icon" />;
-    }
-    return;
+    const iconMap = {
+      notice: {
+        icon: <FiAlertOctagon className="toast-icon" />,
+        backgroundColor: "#D69AFF",
+      },
+      warning: {
+        icon: <FiTool className="toast-icon" />,
+        backgroundColor: "#CECE4A",
+      },
+      success: {
+        icon: <FiCheckCircle className="toast-icon" />,
+        backgroundColor: "#5FD65F",
+      },
+      error: {
+        icon: <FiAlertTriangle className="toast-icon" />,
+        backgroundColor: "#BB4D4D",
+      },
+      default: {
+        icon: <FiAlertCircle className="toast-icon" />,
+        backgroundColor: "#333",
+      },
+    };
+
+    const { icon, backgroundColor } =
+      iconMap[message.toastType] || iconMap.default;
+
+    styles.backgroundColor = backgroundColor;
+    return icon;
   };
 
   return (
-    <div className="toast slide-in-animation" style={styles} ref={ref}>
+    <div
+      className={
+        slideOut ? "toast slide-out-animation" : "toast slide-in-animation"
+      }
+      style={styles}
+    >
       <div className="toast-icon-container">{toastIcon()}</div>
```

Comment:
@Meet-Kalani We should avoid execution of function inside return to avoid re-renders